macOS unable to compile with openCL support

### ImageMagick version

7.1.0

### Operating system

MacOS

### Operating system, version and so on

13 (ventura)

### Description

Trying to compile with OpenCL supports on macOS.

I tried to compile manually but got an error. I tried via home-brew. open-cl is disabled but default in this one so I did
```brew edit imagemagick``` and removed the _--disable-opencl_ arg and added _--enable-opencl_
it worked as I got the same error as when building from source

```
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [MagickCore/libMagickCore-7.Q16HDRI.la] Error 1
```

### Steps to Reproduce

### from source:
./configure --enable-opencl
make

### homebrew
brew edit imagemagick
remove --disable-opencl flag
add --enable-opencl flag
brew reinstall -s imagemagick

### Images

_No response_

This posting describes the same issue I have, using macOS Monterey. There are more than a dozen OpenCL related reference errors [1] in the configuration code that give the "clang: error: linker command failed" error at the end of the make step. (all new to me so I don't know the terminology). I am able to install ImageMagic both from source and with Homebrew, but, not with the  "--enable-opencl" option. Has anyone seen an understandable way to fix this?

[1] Undefined symbols for architecture x86_64:
  "_clBuildProgram", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clCreateBuffer", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clCreateCommandQueue", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clCreateContext", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o ...............

I can confirm that I get errors when trying to manually "make" the install on Mac OSX Monterey on M1 Mac trying to build IM 7.1.0.53 with OpenCL.  (My delegates are installed using MacPorts, but I build IM from source). It works fine without OpenCL.

Here are my messages at the make stage:

```
Undefined symbols for architecture arm64:
  "_clBuildProgram", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clCreateBuffer", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clCreateCommandQueue", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clCreateContext", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clCreateKernel", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clCreateProgramWithBinary", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clCreateProgramWithSource", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clEnqueueMapBuffer", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clEnqueueNDRangeKernel", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clEnqueueReadBuffer", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clEnqueueUnmapMemObject", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clFinish", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clFlush", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clGetDeviceIDs", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clGetDeviceInfo", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clGetEventInfo", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clGetEventProfilingInfo", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clGetKernelInfo", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clGetPlatformIDs", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clGetPlatformInfo", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clGetProgramBuildInfo", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clGetProgramInfo", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clReleaseCommandQueue", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clReleaseContext", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clReleaseEvent", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clReleaseKernel", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clReleaseMemObject", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clReleaseProgram", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clRetainEvent", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clRetainMemObject", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clSetEventCallback", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clSetKernelArg", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
  "_clWaitForEvents", referenced from:
      _InitializeOpenCL in libMagickCore_7_Q16HDRI_la-opencl.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [MagickCore/libMagickCore-7.Q16HDRI.la] Error 1
make: *** [all] Error 2
```

Here is my configure command:

```
./configure \
--enable-openmp  \
--enable-delegate-build --enable-shared --disable-static --enable-opencl \
--with-modules --with-quantum-depth=16 --without-wmf --with-rsvg --with-fftw \
--disable-silent-rules --disable-dependency-tracking --without-pango \
--with-lqr --with-gslib --with-gs-font-dir=/opt/local/share/ghostscript/fonts/

```

There are many google responses for "Undefined symbols for architecture x86_64: "_clBuildProgram", referenced from:.....: "  
Like this one: https://www.positioniseverything.net/undefined-symbols-for-architecture-x86_64

They are all related to how Xcode builds different OS programs, but very similar type errors.  So far I have not understood any of these references enough to apply to IM.

--- a/configure.ac
+++ b/configure.ac
@@ -483,6 +483,7 @@ fi
 no_cl=yes
 AX_HAVE_OPENCL([C])
 if test "X$no_cl" != 'Xyes'; then
+  CPPFLAGS="$CL_CPPFLAGS $CPPFLAGS"
   MAGICK_FEATURES="OpenCL $MAGICK_FEATURES"
 fi
 

--- a/m4/ax_have_opencl.m4
+++ b/m4/ax_have_opencl.m4
@@ -44,7 +44,7 @@ AC_DEFUN([AX_HAVE_OPENCL],
   [enable_opencl=$enableval],
   [enable_opencl='no'])
   if test x"$enable_opencl" != xno ; then
-    CPPFLAGS_CL=""
+    CL_CPPFLAGS=""
     AC_CHECK_HEADERS([CL/cl.h OpenCL/cl.h], [HAVE_CL_H="yes"; break], [HAVE_CL_H="no"])
     if test x"$HAVE_CL_H" = xno ; then
       # check for AMD's SDK
@@ -52,7 +52,7 @@ AC_DEFUN([AX_HAVE_OPENCL],
       if test -d /opt/AMDAPP/include/CL ; then
         HAVE_CL_H="yes"
         AC_DEFINE([HAVE_CL_CL_H])
-        CPPFLAGS_CL="-I/opt/AMDAPP/include"
+        CL_CPPFLAGS="-I/opt/AMDAPP/include"
       fi
       AC_MSG_RESULT([$HAVE_CL_H])
     fi
@@ -62,19 +62,19 @@ AC_DEFUN([AX_HAVE_OPENCL],
       if test -d /usr/local/cuda/include/CL ; then
         HAVE_CL_H="yes"
         AC_DEFINE([HAVE_CL_CL_H])
-        CPPFLAGS_CL="-I/usr/local/cuda/include"
+        CL_CPPFLAGS="-I/usr/local/cuda/include"
       elif test -d /usr/local/cuda-6.5/include/CL ; then
         HAVE_CL_H="yes"
         AC_DEFINE([HAVE_CL_CL_H])
-        CPPFLAGS_CL="-I/usr/local/cuda-6.5/include"
+        CL_CPPFLAGS="-I/usr/local/cuda-6.5/include"
       elif test -d /usr/local/cuda-6.0/include/CL ; then
         HAVE_CL_H="yes"
         AC_DEFINE([HAVE_CL_CL_H])
-        CPPFLAGS_CL="-I/usr/local/cuda-6.0/include"
+        CL_CPPFLAGS="-I/usr/local/cuda-6.0/include"
       elif test -d /usr/local/cuda-5.5/include/CL ; then
         HAVE_CL_H="yes"
         AC_DEFINE([HAVE_CL_CL_H])
-        CPPFLAGS_CL="-I/usr/local/cuda-5.5/include"
+        CL_CPPFLAGS="-I/usr/local/cuda-5.5/include"
       fi
       AC_MSG_RESULT([$HAVE_CL_H])
     fi
@@ -87,12 +87,12 @@ AC_DEFUN([AX_HAVE_OPENCL],
       #
       # First we check for Mac OS X, since OpenCL is standard there
       #
-      LIBS_CL="none"
+      CL_LIBS=""
       AC_MSG_CHECKING([for OpenCL library])
       case "$host_os" in
         darwin*) # On Mac OS X we check for installed frameworks
         AX_CHECK_FRAMEWORK([OpenCL], [
-          LIBS_CL="-framework OpenCL"
+          CL_LIBS="-framework OpenCL"
           no_cl=no
           AC_MSG_RESULT(yes)
           CL_ENABLED=true
@@ -109,7 +109,7 @@ AC_DEFUN([AX_HAVE_OPENCL],
       *)
         save_LIBS=$LIBS
         save_CFLAGS=$CFLAGS
-        CFLAGS=$CPPFLAGS_CL
+        CFLAGS=$CL_CPPFLAGS
         LIBS="$save_LIBS -L/usr/lib64/nvidia -L/usr/lib/nvidia -lOpenCL"
         AC_LINK_IFELSE(
         [AC_LANG_PROGRAM([
@@ -126,7 +126,7 @@ AC_DEFUN([AX_HAVE_OPENCL],
           AC_MSG_RESULT(yes)
           CL_ENABLED=true
           CL_VERSION=1
-          LIBS_CL="$save_LIBS -L/usr/lib64/nvidia -L/usr/lib/nvidia -lOpenCL"
+          CL_LIBS="-L/usr/lib64/nvidia -L/usr/lib/nvidia -lOpenCL"
         ],[
           no_cl=yes
           AC_MSG_RESULT(no)

